### PR TITLE
Timeline charts example

### PR DESCRIPTION
**Pull request by kristinamongo**
This PR is adding the code for the timeline chart example using the Charts Embedding SDK.

I've created the app with create-react-app but tried to remove some of the fluff that doesn't serve.

The readme is trying to explain what this is. If you think it's not clear, please advise.

I've tried to keep the packages dependency low so the example is lightweight and easy to understand.
The example is rewritten to use react hooks so that there are no additional renders. 
All suggestions on how this can be more efficient are highly welcome.

### CODE REVIEW

File: examples/timeline-charts-example/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "timeline-chart",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@material-ui/core": "^4.9.12",
+    "@mongodb-js/charts-embed-dom": "1.1.2",
```
Comment (scottsidwell) [inline examples/timeline-charts-example/package.json:7]: I'd change this to: `^1.1.2`

File: examples/timeline-charts-example/public/index.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
```
Comment (scottsidwell) [inline examples/timeline-charts-example/public/index.html:5]: I'd just remove the favicon, it's unnecessary 🤷

File: examples/timeline-charts-example/src/Dashboard.jsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+const columnChart = sdk.createChart({
+  chartId: "ff518bbb-923c-4c2c-91f5-4a2b3137f312", // Optional: ~REPLACE~ with the Chart ID from your Embed Chart dialog
+});
+
+const geoChart = sdk.createChart({
+  chartId: "b1983061-ee44-40ad-9c45-4bb1d4e74884", // Optional: ~REPLACE~ with the Chart ID from your Embed Chart dialog
+});
+
+const PrettoSlider = withStyles({
```
Comment (scottsidwell) [inline examples/timeline-charts-example/src/Dashboard.jsx:27]: typo: `PrettySlider`

Could consider changing the name to: TimelineSlider

Comment (scottsidwell) [reply examples/timeline-charts-example/src/Dashboard.jsx:27]: I'd also move this out into a separate file and just `import PrettySlider from './PrettySlider'`

File: examples/timeline-charts-example/src/Dashboard.jsx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+import Slider from "@material-ui/core/Slider";
+import ChartsEmbedSDK from "@mongodb-js/charts-embed-dom";
+import { useEffect } from "react";
+
+const firstOlympicsYear = 1896;
+const lastOlympicsYear = 2016;
+const timelineInterval = 2000; //ms
+
+let timerId;
+let replay = false;
```
Comment (scottsidwell) [inline examples/timeline-charts-example/src/Dashboard.jsx:13]: These variables should be defined and referenced inside the `Dashboard` component via refs
e.g.
```
const timerIdRef = React.useRef();
const replayRef = React.useRef(false);

...
timerIdRef.current = setInterval(() => {})
clearInterval(timerIdRef.current);
...
replayRef.current = true;
```

Comment (kristinamongo) [reply examples/timeline-charts-example/src/Dashboard.jsx]: Good idea, thanks :)

File: examples/timeline-charts-example/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+## Background
+
+📄 _[See the MongoDB Charts Embedding Docs for more details](https://docs.mongodb.com/charts/saas/embedding-charts/)_
+
+The example code in this directory is building a small react app, implementing a chart timeline using the Charts Embedding SDK.
+
+What the application is doing is showing the distribution of Olympic medals through the years.
+
+We are changing the filter on the data every few seconds so that it can look like a timeline video.
```
Comment (jarjee) [inline examples/timeline-charts-example/README.md:11]:
```suggestion
We are using a [filter](https://docs.mongodb.com/charts/saas/filter-embedded-charts/#filter-data-on-charts-embedded-with-the-sdk) on the Olympic year, changing the filter every few seconds to turn it into a timelapse.
```

File: examples/timeline-charts-example/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+The idea is this - for every Olympic year we are filtering all data from the beginning of the Olympic games to the current year.
```
Comment (jarjee) [inline examples/timeline-charts-example/README.md:13]:
```suggestion
We are using a [filter](https://docs.mongodb.com/charts/saas/filter-embedded-charts/#filter-data-on-charts-embedded-with-the-sdk) on the Olympic year, changing the filter every few seconds to turn it into a timelapse.
```

File: examples/timeline-charts-example/.gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+/node_modules
```
Comment (jarjee) [inline examples/timeline-charts-example/.gitignore:1]: This should either be:
- Deleted (this folder isn't a git repo)
- Moved to the root of the git repo

File: examples/timeline-charts-example/README.md (hunk 6 above)
Comment (kristinamongo) [reply examples/timeline-charts-example/README.md:13]: You are right! Thanks, I will change it.

**Pull request merged**